api/api-config-repo-operations-v1/src/main/java/com/thoughtworks/go/apiv1/configrepooperations/ConfigRepoOperationsControllerV1.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2018 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.apiv1.configrepooperations;
+
+import com.google.gson.Gson;
+import com.thoughtworks.go.api.ApiController;
+import com.thoughtworks.go.api.ApiVersion;
+import com.thoughtworks.go.api.spring.ApiAuthenticationHelper;
+import com.thoughtworks.go.api.util.HaltApiResponses;
+import com.thoughtworks.go.api.util.MessageJson;
+import com.thoughtworks.go.apiv1.configrepooperations.representers.PartialConfigParseResultRepresenter;
+import com.thoughtworks.go.config.GoRepoConfigDataSource;
+import com.thoughtworks.go.config.PartialConfigParseResult;
+import com.thoughtworks.go.config.remote.ConfigRepoConfig;
+import com.thoughtworks.go.domain.materials.MaterialConfig;
+import com.thoughtworks.go.server.materials.MaterialUpdateService;
+import com.thoughtworks.go.server.service.ConfigRepoService;
+import com.thoughtworks.go.server.service.MaterialConfigConverter;
+import com.thoughtworks.go.spark.Routes.ConfigRepos;
+import com.thoughtworks.go.spark.spring.SparkSpringController;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import spark.Request;
+import spark.Response;
+
+import java.util.Collections;
+
+import static spark.Spark.*;
+
+@Component
+public class ConfigRepoOperationsControllerV1 extends ApiController implements SparkSpringController {
+
+    private static final Gson GSON = new Gson();
```

Comment:
We've intentionally stayed away from using GSON for serialization/deserialization of responses. For performance reasons.